File: MAP(Advanced Programming Methods)/lab5/src/Model/Statement/CompoundStatement.java
package Model.Statement;
import Model.*;
import Model.Expression.Expression;
import Model.ADT.*;
public class CompoundStatement implements Statement {

    private Statement first,second;

    public CompoundStatement(Statement f,Statement s)
    {
        this.first=f;
        this.second=s;
    }

    @Override
    public PrgState execute(PrgState p)
    {
        p.getExeStack().push(second);
        p.getExeStack().push(first);
        return p;
    }

    @Override
    public String toString()
    {
        StringBuffer buff = new StringBuffer();
        buff.append("{\n\t");
        buff.append(this.first);
        buff.append("\n\t");
        buff.append(this.second);
        buff.append("\n}");
        //return "{\n"+this.first + "\n"+ this.second + "\n" +"}\n";
        return buff.toString();
    }
}

File: MAP(Advanced Programming Methods)/LabAssig7/LabAssig7/Model/Statements/CompoundStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LabAssig7.Model.Statements
{
    public class CompoundStatement:Statement
    {
        private Statement first, second;

        public CompoundStatement(Statement f,Statement s)
        {
            this.first = f;
            this.second = s;
        }

        public PrgState Execute(PrgState p)
        {
            p.Stack.Push(this.second);
            p.Stack.Push(this.first);
            return p;
        }

        public override string ToString()
        {
            StringBuilder buff = new StringBuilder();
            buff.Append("{");
            buff.Append(this.first);
            buff.Append(this.second);
            buff.Append("}\n");
            return buff.ToString();

        }
    }
}
